packages/components/time-picker/src/common/picker.vue

```diff
@@ -557,8 +557,17 @@ export default defineComponent({
       ctx.emit('calendar-change', e)
     }
 
+    const baseProps: any = ref({ ...props })
+    watch(
+      () => props.shortcuts,
+      (val) => {
+        baseProps.value.shortcuts = val
+      },
+      { immediate: true }
+    )
+
```

Comment:
First:

Do not destruct `props` which will lose its reactivity,

convert this like

```
import { toRefs } from 'vue'
const baseProps = toRefs(props) // plus do not use any annotation unless you cannot work around.
```

2. With `toRef` method, `baseProps` is naturally reactive, refer to: https://v3.vuejs.org/api/refs-api.html#torefs
You do not have to watch `props.shortcuts` because THEY WERE CONNECTED.